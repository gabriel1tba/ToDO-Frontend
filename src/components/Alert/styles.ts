import styled, { css } from 'styled-components';

export const Overlay = styled.div`
  z-index: 1;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgba(0, 0, 0, 0.6);
  backdrop-filter: blur(3px);
  padding: 1rem;
`;

export const Wrapper = styled.div`
  ${({ theme }) => css`
    z-index: 2;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 2rem;
    gap: 1.5rem;

    width: 432px;
    height: 238px;

    background: ${theme.colors.background};
    border-radius: 8px;

    color: ${theme.colors.gray[500]};

    h1 {
      font-size: 1.5rem;
      line-height: 2rem;
      font-weight: 700;
    }

    p {
      font-size: 1rem;
      line-height: 1.5rem;
      font-weight: 400;
      max-width: 240px;
      text-align: center;
    }

    div {
      display: flex;
      gap: 1rem;
    }
  `}
`;
