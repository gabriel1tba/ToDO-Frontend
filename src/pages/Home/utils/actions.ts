import { TTodo } from 'services/TodoService/interfaces';

export enum Types {
  GET_TODOS = 'GET_TODOS',
  CREATE_TODO = 'CREATE_TODO',
  UPDATE_TODO = 'UPDATE_TODO',
  DELETE_TODO = 'DELETE_TODO',
}

interface GetTodos {
  type: Types.GET_TODOS;
  payload: TTodo[];
}

interface CreateTodo {
  type: Types.CREATE_TODO;
  payload: TTodo;
}

interface UpdateTodo {
  type: Types.UPDATE_TODO;
  payload: TTodo;
}

interface DeleteTodo {
  type: Types.DELETE_TODO;
  payload: string;
}

export type Action = GetTodos | CreateTodo | UpdateTodo | DeleteTodo;

export const getTodosAction = (todos: TTodo[]): GetTodos => ({
  type: Types.GET_TODOS,
  payload: todos,
});

export const createTodoAction = (todo: TTodo): CreateTodo => ({
  type: Types.CREATE_TODO,
  payload: todo,
});

export const updateTodoAction = (todo: TTodo): UpdateTodo => ({
  type: Types.UPDATE_TODO,
  payload: todo,
});

export const deleteTodoAction = (id: string): DeleteTodo => ({
  type: Types.DELETE_TODO,
  payload: id,
});
