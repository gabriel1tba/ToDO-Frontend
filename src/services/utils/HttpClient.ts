import axios from 'axios';

const HttpClient = axios.create({
  baseURL: `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}`,
});

export default HttpClient;
