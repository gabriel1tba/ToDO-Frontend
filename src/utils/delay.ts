const delay = (ms = 1000): Promise<unknown> =>
  new Promise((resolve) => setTimeout(resolve, ms));

export default delay;
