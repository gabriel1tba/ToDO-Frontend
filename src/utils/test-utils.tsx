import { render, RenderOptions } from '@testing-library/react';
import { TodoProvider } from 'context/todos';

interface IWrapperProvider {
  children: React.ReactNode;
}

const WrapperProvider = ({ children }: IWrapperProvider) => (
  <TodoProvider>{children}</TodoProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(<WrapperProvider>{ui}</WrapperProvider>, { ...options });

export * from '@testing-library/react';
export { customRender as render };
