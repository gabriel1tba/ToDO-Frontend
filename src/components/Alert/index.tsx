import Button from 'components/Button';

import * as S from './styles';

export interface IAlert {
  title: string;
  description: string;
  isOpen: boolean;
  isLoading?: boolean;
  onClose: () => void;
  onConfirm: () => void;
}

const Alert = ({
  title,
  description,
  isOpen,
  isLoading,
  onClose,
  onConfirm,
}: IAlert) => {
  if (!isOpen) return null;

  return (
    <S.Overlay>
      <S.Wrapper>
        <h1>{title}</h1>
        <p>{description}</p>
        <div>
          <Button disabled={isLoading} color="outline" onClick={onClose}>
            Cancelar
          </Button>
          <Button loading={isLoading} color="danger" onClick={onConfirm}>
            Sim, excluir
          </Button>
        </div>
      </S.Wrapper>
    </S.Overlay>
  );
};

export default Alert;
