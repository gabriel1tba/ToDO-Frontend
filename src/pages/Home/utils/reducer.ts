import { TTodo } from 'services/TodoService/interfaces';

import { Types, Action } from './actions';

const todosReducer = (state: TTodo[], action: Action): TTodo[] => {
  switch (action.type) {
    case Types.GET_TODOS: {
      return action.payload;
    }

    case Types.CREATE_TODO: {
      return [...state, action.payload];
    }

    case Types.UPDATE_TODO: {
      return state.map((todo) => {
        if (action.payload.id === todo.id) {
          todo = { ...action.payload, created_at: todo.created_at };
        }
        return todo;
      });
    }

    case Types.DELETE_TODO: {
      return state.filter((todo) => {
        return todo.id !== action.payload;
      });
    }

    default:
      return state;
  }
};

export default todosReducer;
