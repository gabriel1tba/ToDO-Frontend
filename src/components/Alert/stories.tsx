import { Story, Meta } from '@storybook/react';

import Alert, { IAlert } from '.';

export default {
  title: 'Alert',
  component: Alert,
  argTypes: {
    title: {
      type: 'string',
    },
    description: {
      type: 'string',
    },
    isOpen: {
      type: 'boolean',
    },
  },
} as Meta;

export const Default: Story<IAlert> = (args) => <Alert {...args} />;

Default.args = {
  title: 'Você ainda não tem tarefas cadastradas',
  description: 'Crie tarefas clicando no botão Nova Tarefa',
  isOpen: true,
};
