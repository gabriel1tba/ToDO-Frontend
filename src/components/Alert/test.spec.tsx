import { render, screen } from 'utils/test-utils';

import Alert from '.';

describe('<Alert />', () => {
  beforeEach(() => {
    render(<div id="portal-modal-root" />);
  });

  it('should render the Alert correctly', () => {
    const { container } = render(
      <Alert
        isOpen
        title="Deseja mesmo excluir essa tarefa?"
        description="Essa ação não poderá ser desfeita"
        onCancel={() => null}
        onConfirm={() => null}
      />
    );

    expect(
      screen.getByText(/deseja mesmo excluir essa tarefa?/i)
    ).toBeInTheDocument();
    expect(
      screen.getByText(/essa ação não poderá ser desfeita?/i)
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the Alert loading correctly', () => {
    render(
      <Alert
        isOpen
        isLoading
        title="Deseja mesmo excluir essa tarefa?"
        description="Essa ação não poderá ser desfeita"
        onCancel={() => null}
        onConfirm={() => null}
      />
    );

    expect(screen.getByRole('button', { name: 'Cancelar' })).toBeDisabled();
    expect(screen.getByRole('button', { name: 'Sim, excluir' })).toBeDisabled();
  });
});
