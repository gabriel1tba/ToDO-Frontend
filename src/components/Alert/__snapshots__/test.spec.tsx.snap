// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should render the Alert correctly 1`] = `
.c0 {
  position: absolute;
  right: 0;
  top: 0;
  padding: 30px;
  overflow: hidden;
  z-index: 3;
}

<div
  class="c0"
/>
`;
